File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;

        int minIndex = 0;
        int maxIndex = 0;

        // find min and max indices 
        for (int i = 0; i < n; i++) {
            if (nums[i] < nums[minIndex])
                minIndex = i;
            
            if (nums[i] > nums[maxIndex])
                maxIndex = i;
        }

        int left = Math.min(minIndex, maxIndex);
        int right = Math.max(minIndex, maxIndex);

        // remove both from the front
        int fromFront = right + 1;

        // remove both from the back
        int fromBack = n - left;

        // remove one from front, and one from back
        int fromBoth = (left + 1) + (n - right);

        return Math.min(fromFront, Math.min(fromBack, fromBoth));
    }
}

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: list[int]) -> int:
        n = len(nums)

        minIndex = 0
        maxIndex = 0

        # find min and max indices 
        for i in range(n):
            if nums[i] < nums[minIndex]:
                minIndex = i
            
            if nums[i] > nums[maxIndex]:
                maxIndex = i

        left = min(minIndex, maxIndex)
        right = max(minIndex, maxIndex)

        # remove both from the front
        fromFront = right + 1

        # remove both from the back
        fromBack = n - left

        # remove one from front, and one from back
        fromBoth = (left + 1) + (n - right)

        return min(fromFront, min(fromBack, fromBoth))
